client/client.go

```diff
@@ -48,6 +72,15 @@ func decodeWithNumber(r io.Reader, value interface{}) error {
 
 func unixDialer(socketPath string) func(string, string) (net.Conn, error) {
 	return func(_, _ string) (net.Conn, error) {
+		if _, err := os.Stat(socketPath); os.IsNotExist(err) {
+			return nil, &SocketNotFoundError{
+				Err:        err,
+				SocketPath: socketPath,
+			}
+		} else if err != nil {
+			return nil, fmt.Errorf("checking if pebble socket path %q exists: %w", socketPath, err)
```

Comment:
This error message was a bit confusing to me at first. It should also basically never happen. What about dropping the word "pebble" and tightening it to just `fmt.Errorf("cannot stat %q: %w", socketPath, err)`?

Reply:
Agreed